crates/nu-protocol/src/value/mod.rs

```diff
@@ -1581,82 +1532,96 @@ impl Value {
                             let v_span = val.span();
                             match val {
                                 Value::Record { val: record, .. } => {
-                                    for (col, val) in record.iter_mut() {
-                                        if col == col_name {
-                                            if cell_path.len() == 1 {
-                                                return Err(ShellError::ColumnAlreadyExists {
-                                                    col_name: col_name.to_string(),
-                                                    span: *span,
-                                                    src_span: v_span,
-                                                });
-                                            } else {
-                                                return val.insert_data_at_cell_path(
-                                                    &cell_path[1..],
-                                                    new_val,
-                                                    head_span,
-                                                );
-                                            }
+                                    if let Some(val) = record.get_mut(col_name) {
+                                        if path.is_empty() {
+                                            return Err(ShellError::ColumnAlreadyExists {
+                                                col_name: col_name.clone(),
+                                                span: *span,
+                                                src_span: v_span,
+                                            });
+                                        } else {
+                                            val.insert_data_at_cell_path(
+                                                path,
+                                                new_val.clone(),
```

Comment:
Why do we now need a clone here?

Reply:
It's in a loop, so unfortunately the function may be called multiple times and so needs a clone.

Reply:
Thanks for the explanation. 
Resolved for me.